docs/contributing.md

```diff
@@ -200,6 +200,110 @@ resource is one which is expected to hold multiple resources of the same type, b
 different names. If these resources are required to be updated, then an `Update()` call is
 necessary on each one as can be seen above.
 
+### Handlers and Watching
+This file contains the entrypoints into the watch functions that are used in Clowder. Watches are
+used to get Clowder to reconcile when another resource changes. Clowder creates a number of
+resources, and example of which is a `Deployment`. When a Deployment that Clowder creates changes,
+Clowder needs to know about it so that it can reconcile again and replace the changes if necessary.
+Below is an example of multiple watches being set up in the controller.
+
+```golang
+watchers := []Watcher{
+    {obj: &apps.Deployment{}, filter: deploymentFilter},
+    {obj: &core.Service{}, filter: generationOnlyFilter},
+    {obj: &core.ConfigMap{}, filter: generationOnlyFilter},
+    {obj: &core.Secret{}, filter: alwaysFilter},
+}
+
+for _, watcher := range watchers {
+    err := r.setupWatch(ctrlr, mgr, watcher.obj, watcher.filter)
+    if err != nil {
+        return err
+    }
+}
+```
+
+This example sets up four watcher objects with various different filters. There are multiple levels
+of filtering that happens to ensure Clowder only reconciles when necessary is to employ the use of
+filters. An example of over reconciling would be when a deployment with multiple pods is started, or
+redeploys. Every time a pod becomes available, it will change the Deployment resources. This would
+ordinarily trigger a reconciliation of ClowdApp, or ClowdEnvironment that owns it. With the correct
+filtering in place that doesn't happen.
+
+The `generationOnlyFilter` looks like this:
+
+```golang
+func deploymentFilter(logr logr.Logger, ctrlName string) HandlerFuncs {
+	return genFilterFunc(deploymentUpdateFunc, logr, ctrlName)
+}
+```
+
+This creates a new `HandlerFuncs` object that has been configured with the `deploymentUpdateFunc`
+object.
+
+```golang
+func deploymentUpdateFunc(e event.UpdateEvent) bool {
+	objOld := e.ObjectOld.(*apps.Deployment)
+	objNew := e.ObjectNew.(*apps.Deployment)
```

Comment:
For lines 246 and 247, doesn't this just set the dame Deployment object to both `objOld` and `objNew`?